src/Run/Setup.cs

```diff
@@ -14,13 +18,51 @@ public class Setup
         public string Os { get; set; }
         public string ConfigurationFilePath { get; set; }
 
+        private string ParseSettingValue(string inputValue)
+        {
+            string value = string.Empty;
+            int length = inputValue.Length;
+            for (int i = 0; i < length; i++)
+            {
+                if (inputValue[i] == '$')
+                {
+                    if (inputValue[i + 1] == '{')
+                    {
+                        int j;
+                        string memberName = string.Empty;
+                        for (j = i + 2; inputValue[j] != '}' && j < length; j++)
+                            memberName += inputValue[j];
+
+                        // The string is not of format ${}, just add the chars to the value.
+                        if (j == length)
+                            value += "${" + memberName;
+                        else
+                            value += SettingValueProvider.Get(memberName);
+
+                        // Put i to j counter.
+                        i = j;
+                    }
+                    else
+                    {
+                        // If next char is not { then add $ to the value.
+                        value += inputValue[i];
+                    }
+                }
+                else
+                {
+                    value += inputValue[i];
+                }
+            }
+
+            return value;
+        }
 
         private string FindSettingValue(string valueToFind)
         {
             Setting value;
             if (Settings.TryGetValue(valueToFind, out value))
             {
-                return value.DefaultValue;
+                return ParseSettingValue(value.DefaultValue);
```

Comment:
yeah, my bad. I meant in the `default value` field, but after our conversation I understand why you don't want that constrain